### PR TITLE
{Network} `az network application-gateway waf-policy`: Migrate commands to codegen v2

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network application-gateway waf-policy
az network application-gateway rewrite-rule

**Description**

Resolves: https://github.com/Azure/azure-cli/issues/24707
Fixes: https://github.com/Azure/azure-cli/issues/23128

Total: 29 commands

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 1]
```diff
@@ -2356,11 +2356,11 @@ def test_network_app_gateway_waf_policy_setting(self, resource_group):
...
-                     self.check('policySettings.fileUploadLimitInMb', 64),
-                     self.check('policySettings.maxRequestBodySizeInKb', 128),
-                     self.check('policySettings.mode', 'Prevention'),
-                     self.check('policySettings.requestBodyCheck', False),
-                     self.check('policySettings.state', 'Enabled')
+                     self.check('fileUploadLimitInMb', 64),
+                     self.check('maxRequestBodySizeInKb', 128),
+                     self.check('mode', 'Prevention'),
+                     self.check('requestBodyCheck', False),
+                     self.check('state', 'Enabled')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:2363]: breaking change in output?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -296,6 +299,82 @@ def pre_operations(self):
...
+        args_schema.response_header_configurations._registered = False
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if has_value(args.request_headers):
+            configurations = []
+            for k, v in args.request_headers.items():
+                configurations.append({"header_name": k, "header_value": v})
+            args.request_header_configurations = configurations
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:329]: Please use `assign_aaz_list_arg` function in `aaz`

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -1808,56 +1736,45 @@ def list_ag_waf_rule_sets(client, _type=None, version=None, group=None):
...
+            help="Type of the web application firewall rule set.",
+            default="OWASP",
+            enum={"Microsoft_BotManagerRuleSet": "Microsoft_BotManagerRuleSet", "OWASP": "OWASP"},
+        )
+        args_schema.rule_set_version = AAZStrArg(
+            options=["--version"],
+            help="Version of the web application firewall rule set type, 0.1 is used for Microsoft_BotManagerRuleSet",
+            default="3.0",
+            enum={"0.1": "0.1", "2.2.9": "2.2.9", "3.0": "3.0", "3.1": "3.1", "3.2": "3.2"},
+        )
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1755]: Does those argument required or not?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 4]
```diff
@@ -1808,56 +1736,45 @@ def list_ag_waf_rule_sets(client, _type=None, version=None, group=None):
...
-    })
+    def pre_operations(self):
+        args = self.ctx.args
+        managed_rule_set = {
+            "rule_set_type": args.rule_set_type,
+            "rule_set_version": args.rule_set_version
+        }
+        managed_rule_definition = {
+            "managed_rule_sets": [managed_rule_set]
+        }
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:1766]: if not has_value(args.rule_set_type) or not has_value(args.rule_set_version), what's the managed_rule_definition?

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 3 above)
Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:1755]: They're not required, but with default value.

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 4 above)
Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:1766]: There is no case of that as both of them have default value.

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 2 above)
Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:329]: It's a mapping between `AAZDictArg` and `AAZListArg`, how to achieve that?

Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:329]: Good question. Let me think

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**